File: server/app/Models/ProductImage.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class ProductImage extends Model
{
    use HasFactory;

    /**
     * The attributes that are mass assignable.
     *
     * @var list<string>
     */
    protected $fillable = [
        'image_url',
        'image_alt',
        'image_title',
        'product_id',
    ];

    /**
     * Get the product that this image belongs to.
     */
    public function product(): BelongsTo
    {
        return $this->belongsTo(Product::class);
    }
}

File: server/Models/ProductImage.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using server.Models;
namespace server.Models
{
    public class ProductImage
    {
        [Key]
        public long Id { get; set; }

        [Required, MaxLength(255)]
        public string ImageUrl { get; set; } = null!;

        [Required, MaxLength(255)]
        public string ImageAlt { get; set; } = null!;

        [Required, MaxLength(255)]
        public string ImageTitle { get; set; } = null!;

        // FK
        public long ProductId { get; set; }
        public Product Product { get; set; } = null!;

        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }
}
